server/sonar-server/src/main/java/org/sonar/ce/settings/ProjectConfigurationFactory.java

```diff
@@ -36,11 +40,22 @@ public ProjectConfigurationFactory(Settings globalSettings, DbClient dbClient) {
     this.dbClient = dbClient;
   }
 
-  public Configuration newProjectConfiguration(String projectKey) {
+  public Configuration newProjectConfiguration(String projectKey, Optional<Branch> branch) {
+    String effectiveKey = generateProjectKey(projectKey, branch);
     Settings projectSettings = new ProjectSettings(globalSettings);
     dbClient.propertiesDao()
-      .selectProjectProperties(projectKey)
+      .selectProjectProperties(effectiveKey)
       .forEach(property -> projectSettings.setProperty(property.getKey(), property.getValue()));
     return new ConfigurationBridge(projectSettings);
   }
+
+  private static String generateProjectKey(String projectKey, Optional<Branch> branch) {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Refactor this method to not always return the same value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3516)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a "Branch" parameter instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553)